src/services/chat/api.ts

```diff
@@ -0,0 +1,147 @@
+import firestore from '@react-native-firebase/firestore';
+import {
+  DocumentData, DocumentRefData, Query, QuerySnapshot,
+} from '../../types/firebase';
+
+// Collection de chats
+// * Id;
+// * Array de referências de usuários;
+// + * Timestamp da última mensagem (updatedAt);
+
+// Collection de mensagens
+// * Id;
+// * Texto;
+// * Timestamp;
+// * Referência para o usuário que enviou;
+// * Referência para o chat;
+// - * Se visualizado ou não;
+// + * Array de referências para usuários que visualizaram;
+
+const chatDocument = (chatUID: string): DocumentRefData => firestore().collection('chats').doc(chatUID);
+
+const createChat = async (users: Array<DocumentRefData>)
+  : Promise<DocumentData> => {
+  const { FieldValue } = firestore;
+  return (await firestore().collection('chats').add({ users, updatedAt: FieldValue.serverTimestamp() })).get();
+};
+
+const getChat = async (chatUID: string): Promise<DocumentData> => firestore().collection('chats').doc(chatUID).get();
+
+const getOwnChats = (userRef: DocumentRefData)
+  : Query => firestore()
+  .collection('chats')
+  .where('users', 'array-contains', userRef);
+
+const getChatByTarget = async (currentUser: DocumentRefData, targetUser: DocumentRefData): Promise<DocumentData | undefined> => {
+  const query1 = await firestore()
+    .collection('chats')
+    .where('users', '==', [currentUser, targetUser])
```

Comment:
I dislike Firebase and understand if no one wants to change this, but I wonder if `.where('users', 'in', [[currentUser, targetUser], [targetUser, currentUser]])` would work instead of 2 separate queries.

Reply:
Let's test

Reply:
It works. Nice!